src/Compilers/Test/Utilities/CSharp/Extensions.cs

```diff
@@ -163,25 +163,52 @@ private static ImmutableArray<string> SplitMemberName(string name)
         }
 
         public static Symbol GetMember(this CSharpCompilation compilation, string qualifiedName)
+            => compilation.GlobalNamespace.GetMember(qualifiedName);
+
+        public static ISymbol GetMember(this Compilation compilation, string qualifiedName)
+            => compilation.GlobalNamespace.GetMember(qualifiedName);
+
+        public static T GetMember<T>(this CSharpCompilation compilation, string qualifiedName) where T : Symbol
+            => (T)compilation.GlobalNamespace.GetMember(qualifiedName);
+
+        public static T GetMember<T>(this Compilation compilation, string qualifiedName) where T : ISymbol
+            => (T)compilation.GlobalNamespace.GetMember(qualifiedName);
+
+        public static IMethodSymbol GetCopyConstructor(this Compilation compilation, string qualifiedTypeName)
         {
-            return compilation.GlobalNamespace.GetMember(qualifiedName);
+            var type = compilation.GetMember<INamedTypeSymbol>(qualifiedTypeName);
+            return type.InstanceConstructors.Single(c => c.Parameters is [{ Type: var parameterType }] && parameterType.Equals(type, SymbolEqualityComparer.Default));
         }
 
-        public static ISymbol GetMember(this Compilation compilation, string qualifiedName)
+        public static IMethodSymbol GetPrimaryConstructor(this Compilation compilation, string qualifiedTypeName)
         {
-            return compilation.GlobalNamespace.GetMember(qualifiedName);
+            var type = compilation.GetMember<INamedTypeSymbol>(qualifiedTypeName);
+            return type.InstanceConstructors.Single(c => c.DeclaringSyntaxReferences.Any(r => r.GetSyntax() is TypeDeclarationSyntax));
         }
 
-        public static T GetMember<T>(this CSharpCompilation compilation, string qualifiedName) where T : Symbol
+        public static IMethodSymbol GetParameterlessConstructor(this Compilation compilation, string qualifiedTypeName)
         {
-            return (T)compilation.GlobalNamespace.GetMember(qualifiedName);
+            var type = compilation.GetMember<INamedTypeSymbol>(qualifiedTypeName);
+            return type.InstanceConstructors.Single(c => c.Parameters is []);
         }
 
-        public static T GetMember<T>(this Compilation compilation, string qualifiedName) where T : ISymbol
+        public static IMethodSymbol GetIEquatableEquals(this Compilation compilation, string qualifiedTypeName)
         {
-            return (T)compilation.GlobalNamespace.GetMember(qualifiedName);
+            var type = compilation.GetMember<INamedTypeSymbol>(qualifiedTypeName);
+            return type.GetMembers("Equals").OfType<IMethodSymbol>().Single(m => m.Parameters is [{ Type: var parameterType }] && parameterType.Equals(type, SymbolEqualityComparer.Default));
```

Comment:
>return type.GetMembers("Equals").OfType<IMethodSymbol>().Single(m => m.Parameters is [{ Type: var parameterType }] && parameterType.Equals(type, SymbolEqualityComparer.Default)); [](http://example.com/codeflow?start=12&length=178)

This check doesn't guarantee that the method implements an interface member. Perhaps the name of the helper shouldn't imply that it has.

Reply:
What would be a better name?

Reply:
> What would be a better name?

"GetSpecializedEqualsOverload"?